### PR TITLE
Adds TGUI say and TGUI input.

### PR DESCRIPTION
**Pull request by NonQueueingMatt**
- experiment: "Added TGUI Say and TGUI Input, along with some preferences (TGUI Input can be disabled, and the buttons customized a little)."
  - rscadd: "Receiving a good amount of damage (25 pain or 15 normal) will cause you to forcefully say whatever's in your chat window. Better type that help message quickly!"
  - rscadd: "Typing indicators now show when your chat box is open without typing, and when you're actively typing."

### CODE REVIEW

**Comment by alsoandanswer** [comment]
holy mother of based

**Comment by NonQueueingMatt** [comment]
!review

File: code/modules/mob/living/carbon/human/species/species.dm [hunk 1]
```diff
@@ -294,7 +293,7 @@
 	var/list/alterable_internal_organs = list(BP_HEART, BP_EYES, BP_LUNGS, BP_LIVER, BP_KIDNEYS, BP_STOMACH, BP_APPENDIX) //what internal organs can be changed in character setup
 	var/list/possible_external_organs_modifications = list("Normal","Amputated","Prosthesis")
 	/// These are the prefixes of the icon states in talk.dmi.
-	var/list/possible_speech_bubble_types = list("normal")
+	var/list/possible_speech_bubble_types = list("default")
```
Comment (Arrow768) [inline code/modules/mob/living/carbon/human/species/species.dm:296]: SQL query to update that?

File: code/_helpers/text.dm [hunk 2]
```diff
@@ -85,10 +85,10 @@
 #define SYMBOLS_DETECTED_NEW_WORD 5 // symbols that we will interpret as the start of a new word
 
 /**
-  * Filters out undesirable characters from names.
-  *
-  * * allow_numbers - allows numbers and common special characters - used for silicon/other weird things names
-  */
+	* Filters out undesirable characters from names.
+	*
+	* * allow_numbers - allows numbers and common special characters - used for silicon/other weird things names
```
Comment (Generalcamo) [inline code/_helpers/text.dm:90]: I don't believe DMDoc allows tabs here

File: code/datums/mind.dm [hunk 3]
```diff
@@ -441,7 +441,8 @@
 		current.set_psi_rank(text2num(href_list["set_psi_rank"]))
 		return
 	else if(href_list["set_psi_rank_limitless"])
-		var/sure = input(usr, "Limitless is INTENTIONALLY STUPIDLY OVERPOWERED! YOU SHOULD NOT BE USING THIS WITHOUT KNOWING EXACTLY WHAT YOU'RE DOING!", "Don't Get A Staff Complaint") as anything in list("I know what I'm doing!", "I fear no man. But that thing... it scares me!")
+		var/sure = tgui_input_list(usr, "Limitless is INTENTIONALLY STUPIDLY OVERPOWERED! YOU SHOULD NOT BE USING THIS WITHOUT KNOWING EXACTLY WHAT YOU'RE DOING!", \
```
Comment (Generalcamo) [inline code/datums/mind.dm:444]: This should probably be `tgui_alert`

File: code/game/objects/structures/watercloset.dm [hunk 4]
```diff
@@ -410,7 +410,7 @@
 	set name = "Set transfer amount"
 	set category = "Object"
 	set src in view(1)
-	var/N = input("Amount per transfer from this:","[src]") as null|anything in possible_transfer_amounts
+	var/N = tgui_input_list(usr, "Set thmount to transfer from this.", "[src]", possible_transfer_amounts)
```
Comment (Generalcamo) [inline code/game/objects/structures/watercloset.dm:413]:
```suggestion
	var/N = tgui_input_list(usr, "Set the amount to transfer from this.", "[src]", possible_transfer_amounts)
```

**Review by Generalcamo — CHANGES_REQUESTED** [review]
First pass review

File: code/_helpers/text.dm (hunk 2 above)
Comment (NonQueueingMatt) [reply code/_helpers/text.dm:90]: the linter disagrees

**Review by alsoandanswer — APPROVED** [review]

**Review by Generalcamo — APPROVED** [review]

**Review by DreamySkrell — APPROVED** [review]

**Comment by Forktress** [comment]
Prosthesis selection currently asks which limb to change then once selected, no further options appear meaning amputations and prosthesis cannot be added to characters.

**Comment by NonQueueingMatt** [comment]
> Prosthesis selection currently asks which limb to change then once selected, no further options appear meaning amputations and prosthesis cannot be added to characters.

Fixed

**Review by Arrow768 — APPROVED** [review]

**Pull request merged**